pkg/client/s3client.go

```diff
@@ -50,6 +50,7 @@ func NewS3Client(c *Client, instanceName, region string) (s3client *S3Client, er
 	var instanceID string
 	svcs, err := c.ResourceClient.ListInstances(controllerv2.ServiceInstanceQuery{
 		Type: "service_instance",
+		Name: instanceName,
```

Comment:
This change to add instanceName to the ListInstances query prevents you from fetching all service_instances. Since the instanceName is known and passed in. With this change the for loop below could be changed, but it will likely just run through one iteration of the loop.